extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -119,10 +119,14 @@ export class WorkspaceService implements IWorkspaceService {
 	 */
 	public async getProjectsInWorkspace(ext?: string, refreshFromDisk: boolean = false): Promise<vscode.Uri[]> {
 
-		if (refreshFromDisk || this.openedProjects.length === 0) { // always check if nothing cached
+		if (refreshFromDisk || this.openedProjects === undefined) { // always check if nothing cached
```

Comment:
undefined || length 0 
OR only undefined?

Reply:
only undefined.  The length === 0 is what was causing the project before, because if you were adding a project to an empty workspace, it would have a legitimate empty list, skip the cache as a result, scan the workspace to find projects, and then think that it already loaded the project you were trying to add.

Reply:
Now, `undefined` is the unset state, and `[]` is a legitimate cached result.